docs/_writing-tests/general-guidelines.md

```diff
@@ -108,6 +108,30 @@ This is not, however, to discourage testing of edge cases or
 interactions between multiple features; such tests are an essential
 part of ensuring interoperability of the web platform.
 
+Tests should pass when the feature under test exposes the expected behavior,
+and they should fail when the feature under test is not implemented or is
+implemented incorrectly. Tests should not rely on unrelated features if doing
+so causes failures in the latest stable release of [Apple
+Safari][apple-safari], [Google Chrome][google-chrome], [Microsoft
+Edge][micosoft-edge], or [Mozilla Firefox][mozilla-firefox]. They should,
+therefore, not rely on any features aside from the one under test unless they
+are supported in all four browsers.
+
+Existing tests can be used as a guide to identify acceptable features. For
+language features that are not used in existing tests, community-maintained
+projects such as [the ECMAScript compatibility tables][es-compat] and
+[caniuse.com][caniuse] provide an overview of basic feature support across the
+browsers listed above.
+
+For JavaScript code that is re-used across many tests (e.g. `testharness.js`
+and the files located in the directory named `common`), language features that
+were introduced in [ECMAScript 2017][es2017] or later should be avoided. This
```

Comment:
This hasn't aged well, imo. The only thing in es2017 I'm aware of that isn't well supported across all major browsers (outside of IE11) is `SharedArrayBuffer`. More importantly, `async` functions are in es2017 and we certainly have a lot of those!

I think this would need to change to at least 2018 if not 2019, and moreso I'm concerned with us forgetting to update this every few years as features become widely available.

Reply:
```suggestion
are relatively new and not yet supported in all current browser engines should not be used. This
```